flink-clients/src/main/java/org/apache/flink/client/deployment/ClusterClientJobClientAdapter.java

```diff
@@ -87,10 +113,19 @@ public JobID getJobID() {
 	}
 
 	@Override
-	public void close() throws Exception {
-		if (shutdownHook != null) {
-			ShutdownHookUtil.removeShutdownHook(shutdownHook, clusterClient.getClass().getSimpleName(), LOG);
+	public final void close() {
+		if (running.compareAndSet(true, false)) {
+			doClose();
 		}
-		this.clusterClient.close();
+	}
+
+	/**
+	 * Method to be override by subclass which contains actual close actions.
```

Comment:
-> "Method to be _overridden_ by subclass which contains actual close actions."

Reply:
👌